src/ansys/sound/core/sound_composer/source_harmonics.py

```diff
@@ -0,0 +1,327 @@
+                f"\t\t{str_control_values}"
+            )
+        else:
+            str_source = "Not set"
+
+        # Source control info.
+        if self.is_source_control_valid():
+            str_source_control = (
+                f"{self.source_control.control.name}\n"
+                f"\tMin: {self.source_control.control.data.min()}\n"
+                f"\tMax: {self.source_control.control.data.max()}\n"
+                f"\tDuration: "
+                f"{self.source_control.control.time_freq_support.time_frequencies.data[-1]} s"
+            )
+        else:
+            str_source_control = "Not set/valid"
+
+        return f"Harmonics source: {str_source}\nSource control: {str_source_control}"
+
+    @property
+    def source_control(self) -> SourceControlTime:
+        """Harmonics source control.
+
+        Contains the control parameter values over time.
+        """
+        return self.__source_control
+
+    @source_control.setter
+    def source_control(self, source_control: SourceControlTime):
+        """Set the source control."""
+        if not (isinstance(source_control, SourceControlTime) or source_control is None):
+            raise PyAnsysSoundException(
+                "Specified source control object must be of type ``SourceControlTime``."
+            )
+        self.__source_control = source_control
+
+    @property
+    def source_harmonics(self) -> FieldsContainer:
+        """Harmonics source data, as a DPF fields container.
+
+        The harmonics source data consists of a series of orders whose levels depend on RPM.
+        """
+        return self.__source_harmonics
+
+    @source_harmonics.setter
+    def source_harmonics(self, source: FieldsContainer):
+        """Set the harmonics source data, from a DPF fields container."""
+        if source is not None:
+            if not isinstance(source, FieldsContainer):
+                raise PyAnsysSoundException(
+                    "Specified harmonics source must be provided as a DPF fields container."
+                )
+
+            if len(source) < 1:
+                raise PyAnsysSoundException(
+                    "Specified harmonics source must contain at least one order."
+                )
+
+            for f_order in source:
+                if len(f_order.data) < 1:
+                    raise PyAnsysSoundException(
+                        "Each order in the specified harmonics source must contain at "
+                        "least one element."
+                    )
+
+            support_data = source.get_support("control_parameter_1")
+            support_properties = support_data.available_field_supported_properties()
+            support_values = support_data.field_support_by_property(support_properties[0])
+            if len(support_values) != len(source):
+                raise PyAnsysSoundException(
+                    "Harmonics source must contain as many order levels as the number of values "
```

Comment:
Harmonics source must contain as many set of order levels as the number of values "
                    "in the associated control parameter (in the provided DPF fields container, "
                    "the number of fields should be the same as the number of values in the "
                    "fields container support

Reply:
done